convert.go

```diff
@@ -0,0 +1,116 @@
+// Package flatfs is a Datastore implementation that stores all
+// objects in a two-level directory structure in the local file
+// system, regardless of the hierarchy of the keys.
+package flatfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/jbenet/go-os-rename"
+)
+
+func Convert(oldPath string, oldShardFunc string, newPath string, newShardFunc string, out io.Writer) error {
+	old, err := New(oldPath, oldShardFunc, false)
+	if err != nil {
+		return err
+	}
+	new, err := New(newPath, newShardFunc, false)
+	if err != nil {
+		return err
+	}
+
+	res, err := old.Query(query.Query{KeysOnly: true})
+	if err != nil {
+		return err
+	}
+
+	if out != nil {
+		fmt.Fprintf(out, "Moving Keys...\n")
+	}
+
+	// first move the keys
+	count := 0
+	for {
+		e, ok := res.NextSync()
+		if !ok {
+			break
+		}
+		if e.Error != nil {
+			return e.Error
+		}
+
+		err := moveKey(old, new, datastore.RawKey(e.Key))
+		if err != nil {
+			return err
+		}
+
+		count++
+		if out != nil && count%10 == 0 {
+			fmt.Fprintf(out, "\r%d keys so far", count)
+		}
+	}
+
+	if out != nil {
+		fmt.Fprintf(out, "\nCleaning Up...\n")
+	}
+
+	// now walk the old top-level directory
+	dir, err := os.Open(old.path)
+	if err != nil {
+		return err
+	}
+	defer dir.Close()
+	names, err := dir.Readdirnames(-1)
+	if err != nil {
+		return err
+	}
+	for _, fn := range names {
+		if fn == "." || fn == ".." {
+			continue
+		}
+		oldPath := filepath.Join(old.path, fn)
+		inf, err := os.Stat(oldPath)
+		if err != nil {
+			return err
+		}
+		if inf.IsDir() || fn == "SHARDING" || fn == "_README" {
+			// if we are a directory are generated file just remove it
+			err := os.Remove(oldPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			// else move it
+			newPath := filepath.Join(new.path, fn)
```

Comment:
Probably nothing, but if there is something in the old datastore I don't to preserve it.  The other option is to error out.  I don't want to remove the old datastore until everything is moved and the directory is empty.